### PR TITLE
[rpc] getblockchaininfo: add size_on_disk, prune_target_size

### PR DESCRIPTION
**Pull request by sickpig**
Fix pruneheight help text.
Move fPruneMode block to match output ordering with help text.
Add functional tests for new fields in getblockchaininfo.

This is a partial port of Core #11367

### CODE REVIEW

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -1138,7 +1144,26 @@ UniValue getblockchaininfo(const UniValue &params, bool fHelp)
...
+    obj.pushKV("size_on_disk", CalculateCurrentUsage());
     obj.pushKV("pruned", fPruneMode);
+    if (fPruneMode)
+    {
+        CBlockIndex *block = chainActive.Tip();
+        assert(block);
+        {
+            READLOCK(cs_mapBlockIndex);
+            assert(block);
+            while (block && block->pprev && (block->pprev->nStatus & BLOCK_HAVE_DATA))
```
Comment (ptschip) [inline src/rpc/blockchain.cpp:1155]: why the second assert(block)?

Comment (sickpig) [reply src/rpc/blockchain.cpp:1155]: a leftover from the port, will remove. Nice catch.

Comment (sickpig) [reply src/rpc/blockchain.cpp:1155]: done and rebased.

**Comment by sickpig** [comment]
erroneously created against `release`

**Pull request closed without merging**